openedx/core/lib/cache_utils.py

```diff
@@ -1,12 +1,50 @@
 """
 Utilities related to caching.
 """
+import collections
 import cPickle as pickle
 import functools
 import zlib
 from xblock.core import XBlock
 
 
+class Memoized(object):
+    """
+    Decorator. Caches a function's return value each time it is called.
+    If called later with the same arguments, the cached value is returned
+    (not reevaluated).
+
+    https://wiki.python.org/moin/PythonDecoratorLibrary#Memoize
+    """
+    def __init__(self, func):
+        self.func = func
+        self.cache = {}
```

Comment:
Also, have you checked with the Open Source team about licensing? This code is copy-pasta from the python wiki, and I don't see any obvious signs of licensing (one way or another). You might be better served by something like https://pypi.python.org/pypi/pylru (at least until we get to python 3 and can use the version in `functools`)